editor/plugins/tiles/tile_map_layer_editor.cpp

```diff
@@ -3627,57 +3632,259 @@ TileMapLayer *TileMapLayerEditor::_get_edited_layer() const {
 	return Object::cast_to<TileMapLayer>(ObjectDB::get_instance(edited_tile_map_layer_id));
 }
 
+void TileMapLayerEditor::_find_tile_map_layers_in_scene(Node *p_current, const Node *p_owner, Vector<TileMapLayer *> &r_list) const {
+	ERR_FAIL_COND(!p_current || !p_owner);
+	if (p_current != p_owner && p_current->get_owner() != p_owner) {
+		return;
+	}
+	TileMapLayer *layer = Object::cast_to<TileMapLayer>(p_current);
+	if (layer) {
+		r_list.append(layer);
+	}
+	for (int i = 0; i < p_current->get_child_count(); i++) {
+		Node *child = p_current->get_child(i);
+		_find_tile_map_layers_in_scene(child, p_owner, r_list);
+	}
+}
+
+void TileMapLayerEditor::_update_tile_map_layers_in_scene_list_cache() {
+	if (!layers_in_scene_list_cache_needs_update) {
+		return;
+	}
+	EditorNode *en = EditorNode::get_singleton();
+	Node *edited_scene_root = en->get_edited_scene();
+	if (!edited_scene_root) {
+		return;
+	}
+
+	tile_map_layers_in_scene_cache.clear();
+	_find_tile_map_layers_in_scene(edited_scene_root, edited_scene_root, tile_map_layers_in_scene_cache);
+	layers_in_scene_list_cache_needs_update = false;
+}
+
+void TileMapLayerEditor::_node_change(Node *node) {
```

Comment:
This has another problem of being called needlessly for every scene change 🤔
Though the idea is good. You could add a variable `layer_list_dirty` and then update the list once it's requested.

Reply:
I mean, I assumed scene changes and node addition/removal would be less annoying than when changing the selected layer. But I can simply remove the deferred call too, as I already protected other functions with a call to `_update_tile_map_layers_in_scene_list_cache()` anyway (just in case one of the function would be called without the cache being ready.

As you prefer, I don't deal with huge scenes with a lot of TileMaps every day, so that's up to you. :sweat_smile:

Reply:
Ah, then looks like removing the deferred call is enough.